src/docs/rust.mdx

```diff
@@ -0,0 +1,323 @@
+
+- `FusedIterator` in all cases unless there is a strong reason not to
+- `DoubleEndedIterator` if reverse iteration is possible
+- `ExactSizeIterator` if the size is known beforehand
+
+If it is exceptionally hard to write a custom iterator, then it can also be a private [newtype](https://doc.rust-lang.org/1.0.0/style/features/types/newtype.html) around a boxed iterator:
+
+```rust
+pub struct FooIter(Box<dyn Iterator<Item = Foo>>);
+
+impl Iterator for FooIter {
+    type Item = Foo;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        self.0.next()
+    }
+}
+```
+
+### Async-await
+
+In general we like to migrate to async-await using code, though much code exists which does not yet do this.  During migration you may need normal functions which return futures, for their signature you should generally prefer `-> impl Future<Output = ...>` over other return types.  This allows the compiler to figure out the details for callsites and avoids having to force things to be pinned or not.  Pinning should only be done explicitly when part of fields in structs or similar.
+
+In **traits** you can not yet use `async fn` ([see this blog post](https://smallcultfollowing.com/babysteps/blog/2019/10/26/async-fn-in-traits-are-hard/)), in this case we have an exception and do the same as the [async-trait crate](https://crates.io/crates/async-trait) does and functions return `-> Pin<Box<dyn Future<Output = ...>>>`.  This allows seamless use in async-await code or in code dealing directly with futures.  Explicitly writing out the signatures is preferable over pulling in the macro syntactic sugar.
+
+### Discourage unwrap()
+
+This may seem obvious, but *almost always* discourage the use of `.unwrap()`. Even if a piece of code is guaranteed not to panic in its current form (because some precondition is met), it might be reused or refactored in a future situation where the precondition does not hold anymore.
+
+
+
+## Development Environment
+
+We use VSCode for development. Each repository contains settings files configuring code style, linters, and useful features. When opening the project for the first time, make sure to "*Install the Recommended Extensions"*, as they will allow editor assist during coding. The probably most up-to-date example can always be found in the [Relay repository](https://github.com/getsentry/relay/blob/master/.vscode/extensions.json):
+
+- *Rust Analyzer:* A fast and feature-packed language server integration that has been developed as successor to RLS.
+- *CodeLLDB:* Debugger based on LLDB, required to debug with the Rust Analyzer extension
+- *Better TOML and Crates:* Support for `Cargo.toml` files and bumping dependency versions
+
+At the time of writing, Rust Analyzer is still under development. Updates are published frequently, so make sure to confirm the update prompt. Some of its default settings may not be ideal, so we recommend the following settings in your User's `settings.json`:
```

Comment:
Is this still true in a nontrivial sense?

Reply:
I don't think so, let's remove this line.

```suggestion
Recommend Rust Analyzer settings in your User's `settings.json`:
```